activerecord/lib/active_record/connection_adapters/abstract/schema_definitions.rb

```diff
@@ -128,7 +128,14 @@ def foreign_key_options
       def columns
         result = [["#{name}_id", type, options]]
         if polymorphic
-          result.unshift(["#{name}_type", :string, polymorphic_options])
+          options = polymorphic_options
+          if index
+            indexable_string_limit = Base.connection.max_indexable_string_limit
```

Comment:
Is there another way to get this value without hitting the connection? Perhaps we can cache it sometime?